stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/core.py

```diff
@@ -91,21 +91,50 @@ async def get_collection(self, collection_id: str, **kwargs) -> Collection:
 
     @overrides
     async def item_collection(
-        self, collection_id: str, limit: int = 10, token: str = None, **kwargs
+        self,
+        collection_id: str,
+        bbox: Optional[List[NumType]] = None,
+        datetime: Union[str, datetime_type, None] = None,
+        limit: int = 10,
+        token: str = None,
+        **kwargs,
     ) -> ItemCollection:
         """Read an item collection from the database."""
         request: Request = kwargs["request"]
-        base_url = str(kwargs["request"].base_url)
+        base_url = str(request.base_url)
+
+        collection = await self.get_collection(
+            collection_id=collection_id, request=request
+        )
+        try:
+            collection_id = collection["id"]
+        except Exception:
+            raise HTTPException(status_code=404, detail="Collection not found")
```

Comment:
```suggestion
        collection_id = collection.get("id")
        if collection_id is None:
            raise HTTPException(status_code=404, detail="Collection not found")
```

This is a personal preference, but try/catch on `Exception` is/should be an anti-pattern when not required.